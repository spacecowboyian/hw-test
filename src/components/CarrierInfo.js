import React from 'react';
import { Row, Col } from 'react-flexbox-grid';
import { css } from 'emotion/macro';

import { spacing, standardBox } from '../Styles';

const carrierInfo = css`
  ${standardBox}
  text-align: left;
  padding-bottom: 0;
`

const carrierInfoName = css`
  font-size: 46px;
  font-weight: 3100;
  color: #003D40;
`

const infoSummary = css`
  display: flex;
`

const infoSummaryTitle = css`
  margin-block-start: 1.33em;
  font-weight: 600;
`

const infoSummaryDescription = css`
  flex-grow: 1;
`

const infoSummaryDescriptionHeader = css`
  font-weight: 500;
  margin-bottom: calc(${spacing}*2)
`
const infoSummaryDescriptionBody = css`
  font-size: 13px;
`

export default function CarrierInfo({ carrier }) {
  const { carrierName, carrierWebsite, carrierWebsiteInfo } = carrier;
  return (
    <Row className={carrierInfo}>
      <Col md={12} lg={6}>
        <h1 className={carrierInfoName}>{carrierName}</h1>
        <dl className={infoSummary}>
          <dt className={infoSummaryTitle}>Website Summary</dt>
          <dd className={infoSummaryDescription}>
            <h4 className={infoSummaryDescriptionHeader}>{carrierWebsite}</h4>
            <p className={infoSummaryDescriptionBody}>
              {carrierWebsiteInfo}
            </p>
          </dd>
        </dl>
      </Col>
    </Row>
  )
}
