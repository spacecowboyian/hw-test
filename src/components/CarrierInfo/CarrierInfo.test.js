import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import CarrierInfo from './CarrierInfo';

afterEach(cleanup);

it('renders correctly', () => {
  const tree = renderer
    .create(<CarrierInfo carrier={{
      carrierName: 'Test Carrier',
      carrierWebsite: 'www.testcarrier.com',
      carrierWebsiteInfo: 'Boy oh boy is there ever info'
    }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
