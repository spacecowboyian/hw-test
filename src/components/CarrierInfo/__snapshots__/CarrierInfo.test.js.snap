// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-gr20k3-standardBox-carrierInfo row"
>
  <div
    className="col-md-12 col-lg-6"
  >
    <h1
      className="css-4fjyaz-carrierInfoName"
    >
      Test Carrier
    </h1>
    <dl
      className="css-ie6ail-infoSummary"
    >
      <dt
        className="css-172yo6n-infoSummaryTitle"
      >
        Website Summary
      </dt>
      <dd
        className="css-1vhsb9-infoSummaryDescription"
      >
        <h4
          className="css-1669r31-infoSummaryDescriptionHeader"
        >
          www.testcarrier.com
        </h4>
        <p
          className="css-ckr3ut-infoSummaryDescriptionBody"
        >
          Boy oh boy is there ever info
        </p>
      </dd>
    </dl>
  </div>
</div>
`;
